lib/rules/no-wait-for-side-effects.ts

```diff
@@ -48,22 +56,59 @@ export default createTestingLibraryRule<Options, MessageIds>({
       );
     }
 
+    function isRenderInVariableDeclaration(node: TSESTree.Node) {
+      return (
+        isVariableDeclaration(node) &&
+        node.declarations.some((declaration) =>
+          helpers.isRenderVariableDeclarator(declaration)
+        )
+      );
+    }
+
+    function isRenderInAssignment(node: TSESTree.Node) {
+      return (
+        isExpressionStatement(node) &&
+        isAssignmentExpression(node.expression) &&
+        helpers.isRenderUtil(getPropertyIdentifierNode(node.expression.right))
+      );
+    }
+
+    function isRenderInImplicitReturnAssignment(node: TSESTree.Node) {
+      return (
+        isAssignmentExpression(node) &&
+        helpers.isRenderUtil(getPropertyIdentifierNode(node.right))
+      );
+    }
+
+    function isRenderInSequenceAssignment(node: TSESTree.Node) {
+      return (
+        isSequenceExpression(node) &&
```

Comment:
I think you'll be able to reuse [this `getStatementCallExpression` method I added for `no-unnecessary-act`](https://github.com/testing-library/eslint-plugin-testing-library/commit/7a2cba1c171b63f112f3515b147334baf958cb27#diff-823fe8bec495554448323badd523e6b09e436e4ad7e77b5ef6add3ba4024cc38R515).

I'm really close to creating my PR. If mine goes first, you can reuse it here. Otherwise, I'll try to refactor these functions with `getStatementCallExpression`.